lib/methwhy.g

```diff
@@ -119,31 +195,52 @@ end);
 ##  <#/GAPDoc>
 ##
 BIND_GLOBAL("ApplicableMethodTypes",function(arg)
-local oper,narg,args,skip,verbos,fams,flags,i,j,methods,flag,flag2,
-      m,nam,val,erg,has,need,isconstructor;
+local oper,narg,args,verbos,fams,flags,i,j,methods,flag,flag2,
+      m,nam,val,has,need,isconstructor,
+      naf,nummeth,valid,applic,numapplic,nr,first,last;
   if Length(arg)<2 or not IsList(arg[2]) or not IsFunction(arg[1]) then
     Error("usage: ApplicableMethodTypes(<opr>,<arglist>[,<verbosity>[,<nr>]])");
   fi;
+  ## process the arguments 
   oper:=arg[1];
   isconstructor:=IS_CONSTRUCTOR(oper);
   args:=arg[2];
-  if Length(arg)>2 then
-    verbos:=arg[3];
-  else
-    verbos:=0;
+  narg:=Length(args);
+  verbos:=0;
+  if Length(arg)>2 and IsPosInt(arg[3]) then
+    verbos:=Minimum(arg[3],4);
   fi;
-  if Length(arg)>3 then
-    if IsInt( arg[4] ) then
-      skip:=arg[4] - 1;
-    else
-      skip:= -1;
+  nr:=1;
+  if Length(arg)>3 then 
+    if IsPosInt(arg[4]) then
+      nr:=arg[4];
+    elif arg[4]="all" then 
+      nr:=0;
     fi;
-    erg:=[];
-  else
-    skip:=0;
   fi;
-  narg:=Length(args);
-
+  ## if not an operation, just return the function 
+  if not IsOperation(oper) then 
+    nam:=FilenameFunc(oper); 
+    if nam=fail then 
+      Print("#I  ",oper," is not recognised as the name of an operation\n"); 
```

Comment:
In general (I know not always) lines starting `#I` are produced using `Info`, not directly. This makes it easy to turn them off.

You could either (for me) make a new Info level, or use (for example) `Info(InfoWarning, 1, oper, " is not recognised as the name of an operation")`.

Also not that in general printing out arguments to a function is often not a good idea, if someone passes in a GAP function for example, printing it will print out the entire function body.